Sk/test transmission

skroiss-idm opened the pull request:
Adding transmission tests & correcting a few bugs in the transmission step:
1. The gamma function was returning nans if scale_gamma = 0, so I now prevent gamma from being exactly equal to 0
2. The alive counts previously added up 0s because they were using results from the current timestep, but results aren't tallied until the end of the timestep. An exception was when the population was squashed which moved individuals into R. 

I also renamed seir_abm.py as model.py and moved faster_tranmsission into model.py.

jonathanhhb commented:
I'm going to test this out locally soon.

jonathanhhb commented:
I think I'd rather go ahead and swap out the old transmission class entirely and swap in the new one from the file, but not everything has to be done in this step. Otherwise, I have no concerns from what I saw and tested.

skroiss-idm commented:
> I think I'd rather go ahead and swap out the old transmission class entirely and swap in the new one from the file, but not everything has to be done in this step. Otherwise, I have no concerns from what I saw and tested.

@jonathanhhb - I renamed seir_abm.py as model.py and moved faster_tranmsission into model.py. All tests passed and linters run
